tests/test_templatetags_slot_fill.py

```diff
@@ -481,98 +698,297 @@ def test_component_without_default_slot_refuses_implicit_fill(self):
         """
         template = Template(template_str)
 
-        with self.assertRaises(TemplateSyntaxError):
-            template.render(Context({}))
+        with self.assertRaisesMessage(
+            TemplateSyntaxError,
+            "Component 'test_comp' passed default fill content (i.e. without explicit 'name' kwarg), "
+            "even though none of its slots is marked as 'default'",
+        ):
+            template.render(Context())
 
-    @parametrize_context_behavior(["django", "isolated"])
-    def test_component_template_cannot_have_multiple_default_slots(self):
-        class BadComponent(Component):
-            def get_template(self, context):
-                template_str: types.django_html = """
-                    {% load django_components %}
-                    <div>
-                        {% slot "icon" %} {% endslot default %}
-                        {% slot "description" %} {% endslot default %}
-                    </div>
-                """
-                return Template(template_str)
 
-        c = BadComponent("name")
+class PassthroughSlotsTest(BaseTestCase):
+    @parametrize_context_behavior(["isolated", "django"])
+    def test_if_for(self):
+        @register("test")
+        class SlottedComponent(Component):
+            template: types.django_html = """
+                {% load component_tags %}
+                <custom-template>
+                    <header>{% slot "header" %}Default header{% endslot %}</header>
+                    <main>{% slot "main" %}Default main{% endslot %}</main>
+                    <footer>{% slot "footer" %}Default footer{% endslot %}</footer>
+                </custom-template>
+            """
 
-        with self.assertRaises(TemplateSyntaxError):
-            c.render(Context({}))
+            def get_context_data(self, name: Optional[str] = None) -> Dict[str, Any]:
+                return {
+                    "name": name,
+                }
 
-    @parametrize_context_behavior(["django", "isolated"])
-    def test_slot_name_fill_typo_gives_helpful_error_message(self):
```

Comment:
The tradeoff for the changes in this MR is that we can no longer know all slots defined in a component's template ahead of the time. So if a fill with an invalid name is given, we won't raise an error, but simply ignore it.

However, to salvage at least something, when a slot is marked as required, and the corresponding fill is NOT present, then there we still show a helpful message. See https://github.com/EmilStenstrom/django-components/pull/758/files#diff-c7f4341cbd58fb82a995fdba37703267d2dbbdea83ac86c7cb2a67c258fd835cR300